pkg/csi/service/common/commonco/k8sorchestrator/topology.go

```diff
@@ -1517,14 +1532,39 @@ func (volTopology *wcpControllerVolumeTopology) GetSharedDatastoresInTopology(ct
 		}
 
 		// Call GetCandidateDatastores for each cluster moref. Ignore the vsanDirectDatastores for now.
-		for _, clusterMoref := range clusterMorefs {
-			accessibleDs, _, err := cnsvsphere.GetCandidateDatastoresInCluster(ctx, params.Vc, clusterMoref)
+		if !isPodVMOnStretchedSupervisorEnabled {
+			accessibleDs, _, err := cnsvsphere.GetCandidateDatastoresInCluster(ctx, params.Vc, clusterMorefs[0])
 			if err != nil {
 				return nil, logger.LogNewErrorf(log,
 					"failed to find candidate datastores to place volume in cluster %q. Error: %v",
-					clusterMoref, err)
+					clusterMorefs[0], err)
 			}
 			sharedDatastores = append(sharedDatastores, accessibleDs...)
+		} else {
+			var sharedDatastoresForclusterMorefs []*cnsvsphere.DatastoreInfo
+			for index, clusterMoref := range clusterMorefs {
+				accessibleDs, _, err := cnsvsphere.GetCandidateDatastoresInCluster(ctx, params.Vc, clusterMoref)
+				if err != nil {
+					return nil, logger.LogNewErrorf(log,
+						"failed to find candidate datastores to place volume in cluster %q. Error: %v",
+						clusterMoref, err)
+				}
```

Comment:
Can we check for len(accessibleDs) here and error out if it is 0

Reply:
done.

```golang
				if len(accessibleDs) == 0 {
					return nil, logger.LogNewErrorf(log,
						"could not find accessible datastore for the cluster %v", clusterMoref)
				}
```